core/src/OpenMP/Kokkos_OpenMP_Instance.cpp

```diff
@@ -338,17 +338,20 @@ void OpenMPInternal::initialize(int thread_count) {
   }
 
   // Check for over-subscription
+  auto const mpi_local_size = []() {
+    auto ranks = mpi_ranks_per_node();
+    if (ranks < 0) return 1;
+    return ranks;
+  }();
```

Comment:
Why the unusual arrangement of calling a one-off lambda here and in Threads?

Reply:
We have a fair amount of immediately invoked lambdas throughout the code base.  It enables declaring the variable as `const`.  I don't feel strongly about it if you want to suggest changes.

As a side note.  I expect we will eventually refactor the host parallel backends initialization and share more facility between the OpenMP and Threads backends.  The checking for over-subscription is an obvious candidate for that shared facility.